### DISCUSSION TITLE
Publish a blog in Codeberg

### DISCUSSION THREAD

**Discussion by ghost** [quarto-dev/quarto-cli#5488] [Q&A] [answered]
I have published a blog in github without any problem. But when I try with codeberg it does not work.

I have followed the [documentation](https://docs.codeberg.org/codeberg-pages/) of codeberg, but my username.codeberg.page generated website does not work. I would like to know if anyone knows if something different has to be done compared to github. And if possible add it to quarto' s  documentation.

**Comment by mcanouil** [comment]
Could you share the repository on Codeberg?
Did you try to push a simple html file? Does this file is published?

In my opinion and according to the documentation, I don't see a good reason to copy/paste Codeberg documentation as it seems really simple.

```html
<html>

<head>
  <title>Hello World</title>
</head>

<body>
  <h1>Hello World</h1>
</body>

</html>
```

**Reply by mcanouil** [reply, marked as answer]
Only the content of `_site` should be in https://codeberg.org/bambirombi/pages nothing more.
"pages" would be the equivalent of `gh-pages` on `user/user.github.io` repository.
Also, note that I don't know what `Codeberg` pages can do or not, for example, `gh-pages` can process Jekyll files, thus when not using Jekyl you need to add `.nojekyll` file.

**Reply by SHogenboom** [reply]
@mcanouil ; how would I use the `quarto publish gh-pages` command to publish to the `pages` branch that Codeberg needs? It appears to automatically publish to the `gh-pages` branch, and replacing the command to `quarto publish pages` does not work.

**Reply by mcanouil** [reply]
You can't use `quarto publish` as it does not support Codeberg.
You have to use standard transfert protocol such as git checkout / git push, etc.

**Comment by SHogenboom** [comment]
I realise the current implementation doesn’t support Codeberg. But the issue could be easily be resolved with an option to specify the output branch. The only difference is that the branch on Github should be called “gh-pages”, and on Codeberg it should be called “pages”. Everything else works perfectly fine! (I now manually renamed the branch after running the publish command)

I have made this into a feature request: https://github.com/quarto-dev/quarto-cli/discussions/6737
